pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -952,21 +1053,15 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 			return nil
 		}
 
-		//we need this annotation to differentiate between unhealthy machine that
-		//needs remediation, and an unhealthy machine that just got remediated
-		return a.addPoweredOffForRemediationAnnotation(ctx, machine)
-	}
-
-	node, err := a.getNodeByMachine(ctx, machine)
+		node, err := a.getNodeByMachine(ctx, machine)
 
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
-			return err
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
```

Comment:
Wrap and return instead of logging. Also, this message should say what is failing because of not being able to find the machine.

Reply:
I didn't add this code, only moved it. But I can improve it little bit while moving it.